### PR TITLE
Fix 13764: Include assert(decoration != null) in TextField

### PR DESCRIPTION
**Pull request by rubensdemelo**
## Description

Just add an assert to constructor, to avoid `decoration: null` on TextField.

## Related Issues

https://github.com/flutter/flutter/issues/13764

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- `[x]` I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- `[x]` My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- `[x]` All existing and new tests are passing.
- `[x]` I updated/added relevant documentation (doc comments with `///`).
- `[x]` The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- `[x]` I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- `[x]` I signed the [CLA].
- `[x]` I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- `[x]` No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/page_scaffold.dart [hunk 1]
```diff
@@ -96,7 +96,7 @@ class CupertinoPageScaffold extends StatelessWidget {
             viewInsets: newViewInsets,
           ),
           child: Padding(
-            padding: EdgeInsets.only(top: topPadding, bottom: bottomPadding),
+            padding: EdgeInsets.only(bottom: bottomPadding),
```
Comment (dnfield) [inline packages/flutter/lib/src/cupertino/page_scaffold.dart]: This should be in a separate PR - unless it actually is and this is part of the current mixup.

File: packages/flutter/lib/src/cupertino/text_field.dart [hunk 2]
```diff
@@ -177,6 +177,7 @@ class CupertinoTextField extends StatefulWidget {
     this.keyboardAppearance,
     this.scrollPadding = const EdgeInsets.all(20.0),
   }) : assert(textAlign != null),
+       assert(decoration != null),
```
Comment (dnfield) [inline packages/flutter/lib/src/cupertino/text_field.dart]: Can you add a message explaining why this shouldn't be null?

**Comment by HansMuller** [comment]
Something is wrong here. Adding single assert() to TextField should be a one-line change (plus a small test). This PR includes changes to 90 files. 

That said, adding an assert(decoration != null) would be a backwards incompatible change. It's a supported configuration that is tested.  Improving the [Textfield documentation a little](https://docs.flutter.io/flutter/material/TextField/decoration.html) to explain the implications of `decoration: null` would address #13764.

I'd recommend closing this PR and starting again.

**Comment by dnfield** [comment]
This can happen if you try to rebase after opening your PR - merging generally works better and confuses GitHub less.

**Comment by rubensdemelo** [comment]
@HansMuller Probably format code on save change all those files and I miss that before create this PR. Sorry for that. Close and make a new one.
